### PR TITLE
fix(sh-admin): saving changes to server configurations post a failed attempt will require a page reload in dashboard

### PR DESCRIPTION
**Pull request by JoelJacobStephen**
### Ticket
Closes HFE-515

### Description
This PR addresses an issue on the admin dashboard where, when an attempt is made to change server configurations and a mutation error has occurred, after the relevant error toast is displayed, any subsequent attempts to save additional changes to configs fail to trigger the server restart and confirmation modals. This prevents users from making further changes unless they reload the page. This issue is resolved in this PR.

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by nivedin — APPROVED** [review]

File: packages/hoppscotch-sh-admin/src/components/settings/ServerRestart.vue [hunk 1]
```diff
@@ -79,22 +83,22 @@ onMounted(async () => {
...
     );
-    if (!authResult) return;
+    if (!authResult) emit('mutationFailure');
 
     const dataSharingResult = await updateDataSharingConfigs(
       toggleDataSharingMutation
     );
 
-    if (!dataSharingResult) return;
+    if (!dataSharingResult) emit('mutationFailure');
```
Comment (jamesgeorge007) [inline packages/hoppscotch-sh-admin/src/components/settings/ServerRestart.vue:101]: We'll have to return early after emitting the event. Currently, the control flow will proceed with the timer resulting in a page reload after `30s`.

**Review by jamesgeorge007 — CHANGES_REQUESTED** [review]
A failed reset server configuration attempt will still display the modal with the countdown timer.

https://github.com/hoppscotch/hoppscotch/assets/25279263/8c67639a-a1ed-430f-8941-6a4fba2d430a

**Comment by JoelJacobStephen** [comment]
> A failed reset server configuration attempt will still display the modal with the countdown timer.
> 
>  Screen.Recording.2024-05-24.at.6.39.36.PM.mov

@jamesgeorge007 I have resolved the issue

**Review by jamesgeorge007 — APPROVED** [review]
Pushed a [commit](https://github.com/hoppscotch/hoppscotch/pull/4081/commits/05ae344bd46ff83299d40084fceaa3dc09b8024e) ensuring attempts go through after a failed reset configuration action along with some refactoring.

**Pull request merged**